sdl-core/src/main/scala/io/smartdatalake/workflow/SubFeed.scala

```diff
@@ -92,109 +89,102 @@ trait SubFeedConverter[S <: SubFeed] {
    */
-  override def fromSubFeed( subFeed: SubFeed )(implicit context: ActionPipelineContext): SparkSubFeed = {
-    subFeed match {
-      case sparkSubFeed: SparkSubFeed => sparkSubFeed.clearFilter() // make sure there is no filter, as filter can not be passed between actions.
-      case _ => SparkSubFeed(None, subFeed.dataObjectId, subFeed.partitionValues, subFeed.isDAGStart, subFeed.isSkipped)
+  def getDataObjectReadSchema(dataObject: DataObject with CanCreateDataFrame)(implicit context: ActionPipelineContext): Option[GenericSchema] = {
+    dataObject match {
+      case input: UserDefinedSchema if input.schema.isDefined =>
+        input.schema.map(dataObject.createReadSchema)
+      case input: SchemaValidation if input.schemaMin.isDefined =>
+        input.schemaMin.map(dataObject.createReadSchema)
+      case _ => None
     }
   }
+
+  /**
+   * Get an empty DataFrame with a defined schema.
+   * @param dataObjectId Snowpark implementation needs to get the Snowpark-Session from the DataObject. This should not be used otherwise.
+   */
+  def getEmptyDataFrame(schema: GenericSchema, dataObjectId: DataObjectId)(implicit context: ActionPipelineContext): GenericDataFrame
+  def getEmptyStreamingDataFrame(schema: GenericSchema)(implicit context: ActionPipelineContext): GenericDataFrame = throw new NotImplementedError(s"getEmptyStreamingDataFrame is not implemented for ${subFeedType.typeSymbol.name}")
+  def getSubFeed(dataFrame: GenericDataFrame, dataObjectId: DataObjectId, partitionValues: Seq[PartitionValues])(implicit context: ActionPipelineContext): DataFrameSubFeed
+  def createSchema(fields: Seq[GenericField]): GenericSchema
+  def col(colName: String): GenericColumn
+  def lit(value: Any): GenericColumn
+  def max(column: GenericColumn): GenericColumn
+  def explode(column: GenericColumn): GenericColumn
+  /**
+   * Construct array from given columns and removing null values (Snowpark API)
+   */
+  def array_construct_compact(columns: GenericColumn*): GenericColumn
+  def array(columns: GenericColumn*): GenericColumn
+  def struct(columns: GenericColumn*): GenericColumn
+  def expr(sqlExpr: String): GenericColumn
+  def not(column: GenericColumn): GenericColumn
+  def count(column: GenericColumn): GenericColumn
+  def when(condition: GenericColumn, value: GenericColumn): GenericColumn
+  def stringType: GenericDataType
+  def arrayType(dataType: GenericDataType): GenericDataType
+  def structType(colTypes: Map[String,GenericDataType]): GenericDataType
+  /**
+   * Get a DataFrame with the result of the given sql statement.
+   * @param dataObjectId Snowpark implementation needs to get the Snowpark-Session from the DataObject. This should not be used otherwise.
+   */
+  def sql(query: String, dataObjectId: DataObjectId)(implicit context: ActionPipelineContext): GenericDataFrame
+}
+object DataFrameSubFeed {
+  def getHelper(tpe: Type): DataFrameSubFeedCompanion = ScalaUtil.companionOf[DataFrameSubFeedCompanion](tpe)
```

Comment:
From what I have seen, getHelper is only used to be able to use spark/snowparkfunctions. This indicates to me that the split suggested in the previous comment makes sense. Then the Object can be renamed DataFrameFunctions and getHelper can be renamend getFunctions

Reply:
Done, we have now a getCompanion and a getFunctions method